ibis/expr/datatypes.py

```diff
@@ -800,7 +846,7 @@ def name(value):
 _TYPE_PATTERN = re.compile('|'.join(_TYPE_KEYS), flags=re.IGNORECASE)
 
 
-def _generate_tokens(pat, text):
+def _generate_tokens(pat: GenericAny, text: str) -> Iterator[Token]:
```

Comment:
https://docs.python.org/3/library/typing.html#typing.Pattern

Reply:
The issue here is that the scanner method doesn't type check.

Reply:
Yeah, I didn't try it :)